presto-main/src/main/java/com/facebook/presto/operator/BlockEncodingBuffers.java

```diff
@@ -0,0 +1,457 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.facebook.presto.operator;
+
+import com.facebook.presto.spi.block.Block;
+import com.facebook.presto.spi.block.ByteArrayBlock;
+import com.facebook.presto.spi.block.DictionaryBlock;
+import com.facebook.presto.spi.block.Int128ArrayBlock;
+import com.facebook.presto.spi.block.IntArrayBlock;
+import com.facebook.presto.spi.block.LongArrayBlock;
+import com.facebook.presto.spi.block.RunLengthEncodedBlock;
+import com.facebook.presto.spi.block.ShortArrayBlock;
+import io.airlift.slice.SliceOutput;
+
+import javax.annotation.Nullable;
+
+import java.util.Arrays;
+
+import static com.google.common.base.Verify.verify;
+import static io.airlift.slice.SizeOf.SIZE_OF_BYTE;
+import static io.airlift.slice.SizeOf.sizeOf;
+import static java.lang.Math.max;
+import static java.util.Objects.requireNonNull;
+
+public abstract class BlockEncodingBuffers
+{
+    private static final int BITS_IN_BYTE = 8;
+    private static final int INITIAL_POSITION_COUNT = 1024;
+
+    // The initial buffer size in terms of number of elements
+    protected final int initialBufferPositions;
+
+    // The block after peeling off the Dictionary or RLE wrappings.
+    protected Block decodedBlock;
+
+    // The positions (row numbers) of the incoming Block to be copied to this buffer.
+    // All top level BlockEncodingBuffers for the same partition share the same positions array.
+    private int[] positions;
```

Comment:
private and protected member variables are now being mixed together; would you separate them and put private members next to each other and protected members next to each other?